### PR TITLE
Hard Stop Testing / SP1K1 RTD Scaling

### PR DESCRIPTION
**Pull request by nrwslac**
## Description

- Soft Limits updated on SP1K1, MR1K1, MR2/3/4K2.
- Soft Limits swapped on SL1K2 YAG stage.
- RTDs previously unlinked exposed for SP1K1 - RTD order not preserved.

## Motivation and Context

ME exploring and setting hardstops.

## How Has This Been Tested?

This was activated on the PLC. All RTDs checked via function blocks on the PLC.

## Where Has This Been Documented?

https://confluence.slac.stanford.edu/display/L2SI/X-Ray+Optics+Checkout

## Pre-merge checklist
- [ ] Code works interactively
- [ ] Code contains descriptive comments
- [ ] Test suite passes locally
- [ ] Libraries are set to fixed versions and not ``Always Newest``
- [ ] Code committed with ``pre-commit`` (alternatively ``pre-commit run --all-files``)

### CODE REVIEW

File: lcls-plc-rixs-optics/rixs_optics/POUs/PRG_MR1K2_SWITCH.TcPOU [hunk 1]
```diff
@@ -242,7 +242,9 @@ fM1K2_Flow_2(iTermBits:=15, fTermMax:=5.0427, fTermMin:=0.050472);
 fM1K2_Flow_2_val := fM1K2_Flow_2.fReal;
 fM1K2_Press_1(iTermBits:=15, fTermMax:=4.0, fTermMin:=0);
 fM1K2_Press_1_val := fM1K2_Press_1.fReal;
-]]></ST>
+
+main.M5.bLimitBackwardEnable := True;
+main.M5.bLimitForwardEnable := True;]]></ST>
```
Comment (nrwslac) [inline lcls-plc-rixs-optics/rixs_optics/POUs/PRG_MR1K2_SWITCH.TcPOU:247]: limit switches were de-linked, so these need to stay true.

File: lcls-plc-rixs-optics/_Config/NC/Axes/M1K2-Xdwn.xti [hunk 2]
```diff
@@ -1332,37 +1326,32 @@ External Setpoint Generation:
 		</AxisPara>
 		<Encoder Name="Enc" EncType="29">
 			<EncPara ScaleFactorNumerator="1" ScaleFactorDenominator="1000" Offset="-21007.52" MaxCount="#xffffffff">
-				<SoftEndMinControl Enable="true" Range="-1668.52"/>
-				<SoftEndMaxControl Enable="true" Range="2441.7"/>
+				<SoftEndMinControl Range="-1668.52"/>
```
Comment (nrwslac) [inline lcls-plc-rixs-optics/_Config/NC/Axes/M1K2-Xdwn.xti:1329]: should consider whether or not to set these back to true.

File: lcls-plc-rixs-optics/_Config/NC/Axes/M1K1-Yup.xti [hunk 3]
```diff
@@ -1346,7 +1346,7 @@ External Setpoint Generation:
 		<AxisPara>
 			<General UnitName="um"/>
 			<Dynamic Acceleration="5050" Deceleration="5050" Jerk="5100500" DelayTime="0.008"/>
-			<Velo Maximum="500"/>
+			<Velo Maximum="100"/>
```
Comment (nrwslac) [inline lcls-plc-rixs-optics/_Config/NC/Axes/M1K1-Yup.xti:1349]: Gantry runs away from itself moving faster the 100 um/s. I had to decouple the axes twice and fix reduce the gantry error before I figured out that 200 um/s was too fast.

File: lcls-plc-rixs-optics/_Config/NC/Axes/M1K2-Xdwn.xti (hunk 2 above)
Comment (nrwslac) [reply lcls-plc-rixs-optics/_Config/NC/Axes/M1K2-Xdwn.xti:1329]: Secondary axes on other gantries leave the PLC limits false, so I'll leave these false.

File: lcls-plc-rixs-optics/_Config/NC/Axes/M4K2 rX.xti [hunk 4]
```diff
@@ -1325,8 +1325,8 @@ External Setpoint Generation:
 		</AxisPara>
 		<Encoder Name="Enc" EncType="29">
 			<EncPara ScaleFactorNumerator="0.002571" Offset="-130236.02" MaxCount="#xffffffff" ReferenceSystem="1">
-				<SoftEndMinControl Enable="true" Range="-955"/>
-				<SoftEndMaxControl Enable="true" Range="605"/>
+				<SoftEndMinControl Range="-955"/>
+				<SoftEndMaxControl Range="605"/>
```
Comment (nrwslac) [inline lcls-plc-rixs-optics/_Config/NC/Axes/M4K2 rX.xti:1329]: Hardstops still a WIP. Contact Will Lane.

File: lcls-plc-rixs-optics/_Config/NC/Axes/M4K2 rX.xti [hunk 5]
```diff
@@ -1436,7 +1436,7 @@ External Setpoint Generation:
 		</Drive>
 		<Controller Name="Ctrl" CtrType="2">
 			<CtrPara PriorControlFactor="1">
-				<PosDiffControl Range="10"/>
+				<PosDiffControl Range="3.5"/>
```
Comment (nrwslac) [inline lcls-plc-rixs-optics/_Config/NC/Axes/M4K2 rX.xti:1439]: We reduced lag monitoring to try and trip earlier when hitting the hardstops.

File: lcls-plc-rixs-optics/_Config/NC/Axes/M1K1-Pitch.xti [hunk 6]
```diff
@@ -1351,8 +1351,8 @@ External Setpoint Generation:
 		</AxisPara>
 		<Encoder Name="Enc" EncType="29">
 			<EncPara ScaleFactorNumerator="0.00244" Offset="-24484.38252" MaxCount="#xffffffff">
-				<SoftEndMinControl Enable="true" Range="-200"/>
-				<SoftEndMaxControl Enable="true" Range="400"/>
+				<SoftEndMinControl Enable="true" Range="-50"/>
```
Comment (ghalym) [inline lcls-plc-rixs-optics/_Config/NC/Axes/M1K1-Pitch.xti:1354]: Are these documented in a Jira ticket?

File: lcls-plc-rixs-optics/rixs_optics/POUs/PRG_MR1K2_SWITCH.TcPOU (hunk 1 above)
Comment (ghalym) [reply lcls-plc-rixs-optics/rixs_optics/POUs/PRG_MR1K2_SWITCH.TcPOU:247]: Why where they de-linked? is that sage? Do we have Jira #?

**Review by ghalym — COMMENTED** [review]
A lot of soft limits changes. Can you link the Jira or document that has these values?

File: lcls-plc-rixs-optics/_Config/NC/Axes/M1K1-Pitch.xti (hunk 6 above)
Comment (nrwslac) [reply lcls-plc-rixs-optics/_Config/NC/Axes/M1K1-Pitch.xti:1354]: https://jira.slac.stanford.edu/browse/ECS-5028

Comment (nrwslac) [reply lcls-plc-rixs-optics/_Config/NC/Axes/M1K1-Pitch.xti:1354]: Must be a typo, I updated the confluence.

**Review by ghalym — APPROVED** [review]

File: lcls-plc-rixs-optics/rixs_optics/POUs/PRG_MR1K2_SWITCH.TcPOU (hunk 1 above)
Comment (nrwslac) [reply lcls-plc-rixs-optics/rixs_optics/POUs/PRG_MR1K2_SWITCH.TcPOU]: I reverted this change and re-linked the limit switches even though they arent installed; they are wired and sending a signal. https://jira.slac.stanford.edu/browse/ECS-5029

**Pull request merged**